### DISCUSSION TITLE
Programmatically ignore rule violation based on AST inside `.eslintrc.js` file?

### DISCUSSION THREAD

**Discussion by evangalen** [eslint/eslint#17486] [Q&A] [answered]
Would it be possible to programmatically ignore a rule violation based checking the AST from a custom callback function from inside the `.eslintrc.js` file.

In my particular case, I would like to ignore violations for the `no-floating-promises` rule (of `TypeScript-ESLint`) for an invocation of a particular function (e.g. `setFormValue` of Formik).
But I don't want to have to add `eslint-disable-next-line` / `eslint-disable-line` comments everywhere... but instead want to ignore particular function calls.

**Comment by aladdin-add** [comment, marked as answer]
No way to ignore the ast, but you can use `eslint-disable` block. e.g.

```js
/* eslint-disable no-floating-promises */
function setFormValue(){
....
}
/* eslint-enable no-floating-promises */
```

**Reply by evangalen** [reply]
@aladdin-add 
Thanks for your reply!

But having to suppress the rule through comments, even for multiple lines is too cumbersome.
Instead, I would like to be able to whitelist certain functions so that `no-floating-promises` rule just ignores them.

**Reply by aladdin-add** [reply]
well, it sounds like you are proposing a new option for that rule, while the rule was not maintained by eslint team, you can file it in https://github.com/typescript-eslint/typescript-eslint.

**Comment by evangalen** [comment]
At it turns out there is the third-party `eslint-rule-composer` than can be used compose a new rule out of an existing one.
You can then create own `internal-rules` folder in your own repo, and create customized rules in that folder using the `eslint-rule-composer`.

Even though the  `eslint-rule-composer` library has not been maintained for more than 4 year, it still works as a charm.
Hernce, its even used to compose an internal rule in de code base of ESLint itself:
https://github.com/eslint/eslint/blob/f078f9a9e094ec00c61a6ef1c9550d017631e69a/tools/internal-rules/multiline-comment-style.js#L17

Being able to compose a custom internal rule is good enough for me, and therefore I consider my question (self) answered.

**Comment by evangalen** [comment]
Self-answered my question (see above)
